repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/proxy/RequestQueryParamTest.groovy

```diff
@@ -78,48 +78,49 @@ class RequestQueryParamTest extends ReposeValveTest {
-        "/path/to/resource?key=value%5Dvalue"      | "/path/to/resource?key=value%5Dvalue"      | "GET"
-        "/path/to/resource?key=value%20value"      | "/path/to/resource?key=value+value"        | "GET"
-        "/path/to/resource?key=value+value"        | "/path/to/resource?key=value+value"        | "GET"
+        uriSuffixGiven                             | uriSuffixExpected
+        "/path/to/resource?key=value%2Bvalue"      | "/path/to/resource?key=value%2Bvalue"
+        "/path/to/resource?key=value/othervalue"   | "/path/to/resource?key=value%2Fothervalue"
+        "/path/to/resource?key=value:value"        | "/path/to/resource?key=value%3Avalue"
+        "/path/to/resource?key=value@value"        | "/path/to/resource?key=value%40value"
+        "/path/to/resource?key=value?value"        | "/path/to/resource?key=value%3Fvalue"
+        "/path/to/resource?key=value[value"        | "/path/to/resource?key=value%5Bvalue"
+        "/path/to/resource?key=value]value"        | "/path/to/resource?key=value%5Dvalue"
+        "/path/to/resource?key=value%2Fothervalue" | "/path/to/resource?key=value%2Fothervalue"
+        "/path/to/resource?key=value%3Avalue"      | "/path/to/resource?key=value%3Avalue"
+        "/path/to/resource?key=value%40value"      | "/path/to/resource?key=value%40value"
+        "/path/to/resource?key=value%3Fvalue"      | "/path/to/resource?key=value%3Fvalue"
+        "/path/to/resource?key=value%5Bvalue"      | "/path/to/resource?key=value%5Bvalue"
+        "/path/to/resource?key=value%5Dvalue"      | "/path/to/resource?key=value%5Dvalue"
+        "/path/to/resource?key=value%20value"      | "/path/to/resource?key=value+value"
+        "/path/to/resource?key=value+value"        | "/path/to/resource?key=value+value"
     }
 
-    @Unroll("when given a URI that contains space, Repose should reject the request with a 400 - #uriSuffixGiven")
-    def "when given a URI that contains space, Repose should reject the request with a 400"() {
+    @Unroll("when given a URI with a malformed query parameter, Repose should reject the request with a 400 - #uriSuffixGiven")
+    def "when given a URI with a malformed query parameter, Repose should reject the request with a 400"() {
 
         when:
-        MessageChain messageChain = deproxy.makeRequest(url: reposeEndpoint, path: uriSuffixGiven, method: method)
+        MessageChain messageChain = deproxy.makeRequest(url: reposeEndpoint, path: uriSuffixGiven)
 
         then:
         messageChain.handlings.size() == 0
         messageChain.sentRequest.path.endsWith(uriSuffixGiven)
         messageChain.receivedResponse.code == "400"
 
         where:
-        uriSuffixGiven                      | method
-        "/path/to/resource?key=value value" | "GET"
+        uriSuffixGiven << [
+            "/path/to/resource?key=value value",
+            "/path/to/resource?key=value@%",
+            "/path/to/resource?key=va%lu@e"
+        ]
```

Comment:
... and here's the other behavior change.

Reply:
Yep. Between Jetty v9.3.17.v20170317/v9.4.3.v20170317 and v9.3.18.v20170406/v9.4.4.v20170414 the handling of malformed query parameters was changed. Now if it isn't legal, then it doesn't make it in the front door anymore.

 - https://github.com/eclipse/jetty.project/compare/jetty-9.3.17.v20170317...jetty-9.3.18.v20170406
 - https://github.com/eclipse/jetty.project/commit/dff8fb6b90952a92e308d7687bfaf597393876f4

Reply:
Also, if Jetty is validating the query string now, can we remove the following line?
https://github.com/rackerlabs/repose/blob/master/repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java#L401

Reply:
Is that line only intended to vet the query parameters?

Reply:
It is not, which is a fair point.

... but I wonder if Jetty is stricter about URIs in general now. If Jetty validates the entire URI, then this line would no longer be necessary. Should be easy enough to remove this line and run the tests to see what happens, if we are interesting. That can even be done by Jenkins on a separate branch.